Minor ticks and grid lines docs

LiamConnors opened the pull request:
### Documentation PR

- [x] I've [seen the `doc/README.md` file](https://github.com/plotly/plotly.py/blob/master/doc/README.md)
- [x] This change runs in the current version of Plotly on PyPI and targets the `doc-prod` branch OR it targets the `master` branch
- [ ] If this PR modifies the first example in a page or adds a new one, it is a `px` example if at all possible
- [x] Every new/modified example has a descriptive title and motivating sentence or paragraph
- [x] Every new/modified example is independently runnable
- [x] Every new/modified example is optimized for short line count	and focuses on the Plotly/visualization-related aspects of the example rather than the computation required to produce the data being visualized
- [x] Meaningful/relatable datasets are used for all new examples instead of randomly-generated data where possible
- [ ] The random seed is set if using randomly-generated data in new/modified examples
- [ ] New/modified remote datasets are loaded from https://plotly.github.io/datasets and added to https://github.com/plotly/datasets
- [ ] Large computations are avoided in the new/modified examples in favour of loading remote datasets that represent the output of such computations
- [x] Imports are `plotly.graph_objects as go` / `plotly.express as px` / `plotly.io as pio`
- [x] Data frames are always called `df`
- [ ] `fig = <something>` call is high up in each new/modified example (either `px.<something>` or `make_subplots` or `go.Figure`)
- [ ] Liberal use is made of `fig.add_*` and `fig.update_*` rather than `go.Figure(data=..., layout=...)` in every new/modified example
- [x] Specific adders and updaters like `fig.add_shape` and `fig.update_xaxes` are used instead of big `fig.update_layout` calls in every new/modified example
- [x] `fig.show()` is at the end of each new/modified example
- [x] `plotly.plot()` and `plotly.iplot()` are not used in any new/modified example
- [ ] Hex codes for colors are not used in any new/modified example in favour of [these nice ones](https://github.com/plotly/plotly.py/issues/2192)

## Code PR

- [ ] I have read through the [contributing notes](https://github.com/plotly/plotly.py/blob/master/contributing.md) and understand the structure of the package. In particular, if my PR modifies code of `plotly.graph_objects`, my modifications concern the `codegen` files and not generated files.
- [ ] I have added tests (if submitting a new feature or correcting a bug) or
  modified existing tests.
- [ ] For a new feature, I have added documentation examples in an existing or
  new tutorial notebook (please see the doc checklist as well).
- [ ] I have added a CHANGELOG entry if fixing/changing/adding anything substantial.
- [ ] For a new feature or a change in behaviour, I have updated the relevant docstrings in the code to describe the feature or behaviour (please see the doc checklist as well).

-->

nicolaskruchten commented:
Nice! Can you also add an example to the other axes pages like https://plotly.com/python/time-series/ and https://plotly.com/python/log-plot/ (if applicable) ?

nicolaskruchten commented on doc/python/axes.md:
````diff
@@ -380,6 +380,26 @@ fig.update_layout(title_text="Apple Stock Price")
 ```
 
+#### Adding minor ticks
+You can add minor ticks to an axis by specifying with the `minor` property. This takes a `dict` of properties to apply to minor ticks. 
````
please add a line here (and in every new example for this release) in italics with "New in v5.8" or similar :)

LiamConnors replied:
Done

LiamConnors commented:
> Nice! Can you also add an example to the other axes pages like https://plotly.com/python/time-series/ and https://plotly.com/python/log-plot/ (if applicable) ?

I added an example to time-series. It's not supported on log axes. I added a note to say where it isn't supported.

archmoj commented on doc/python/axes.md:
```diff
@@ -388,7 +388,7 @@ You can add minor ticks to an axis with `minor`. This takes a `dict` of properti
 In the following example, we add minor ticks to the x-axis and then to the y-axis. For the y-axis we add ticks on the outside: `minor = {"ticks":"outside"}`. On the x-axis we've spcified some additional properties to style the minor ticks, setting the length of the ticks with `ticklen` and the color with `tickcolor`. We've also turned on grid lines for the x-axis minor ticks using `showgrid`.
 
-Note: Minor ticks and grid lines are not supported on multi-categorical, log, or 3D axes, or on color bars, ternary plots, polar charts, or geo.
+Note: Minor ticks and grid lines are not supported on color bars, ternary plots, polar charts, geo plots, or on multi-categorical, log, or 3D axes.
```
The `log` is actually supported.

LiamConnors replied:
Thanks. I'll add in an example

archmoj replied:
https://github.com/plotly/plotly.js/blob/64a211ba79d07c37c06bc8183cd342f89bc5d1bd/test/jasmine/tests/axes_test.js#L6610-L6623

archmoj replied:
`Note: Minor ticks and grid lines are not supported on ...` 


to


`Note: At the moment minor ticks and grid lines are not supported on ...`

archmoj replied:
`Note: Minor ticks and grid lines are not supported on ...` 


to


`Note: At the moment minor ticks and grid lines are not supported on ...`

archmoj commented on doc/python/axes.md:
```diff
@@ -380,6 +380,30 @@ fig.update_layout(title_text="Apple Stock Price")
+_new in 5.8_
+
+You can add minor ticks to an axis with `minor`. This takes a `dict` of properties to apply to minor ticks. 
+
```
It includes the following properties: 
`tickmode`, `tickvals`, `tickcolor`, `ticklen`, `tickwidth`, `dtick`, `tick0`, `nticks`, `ticks`,
`showgrid`, `gridcolor`, `griddash` and `gridwidth`.

archmoj replied:
It includes the following properties: 
`tickmode`, `tickvals`, `tickcolor`, `ticklen`, `tickwidth`, `dtick`, `tick0`, `nticks`, `ticks`,
`showgrid`, `gridcolor`, `griddash` and `gridwidth`.

archmoj commented on doc/python/axes.md:
```diff
@@ -380,6 +380,30 @@ fig.update_layout(title_text="Apple Stock Price")
+
+_new in 5.8_
+
+You can add minor ticks to an axis with `minor`. This takes a `dict` of properties to apply to minor ticks. 
```
```suggestion
You can position and style minor ticks on a Cartesian axis using `minor`. This takes a `dict` of properties to apply to minor ticks. 
```

LiamConnors replied:
Added

archmoj commented on doc/python/axes.md:
```diff
@@ -380,6 +380,30 @@ fig.update_layout(title_text="Apple Stock Price")
+
+_new in 5.8_
+
+You can add minor ticks to an axis with `minor`. This takes a `dict` of properties to apply to minor ticks. 
```
```suggestion
You can position and style minor ticks on a Cartesian axis using `minor`. This takes a `dict` of properties to apply to minor ticks. 
```

archmoj commented on doc/python/axes.md:
```diff
@@ -380,6 +380,30 @@ fig.update_layout(title_text="Apple Stock Price")
+
+In the following example, we add minor ticks to the x-axis and then to the y-axis. For the y-axis we add ticks on the outside: `minor = {"ticks":"outside"}`. On the x-axis we've spcified some additional properties to style the minor ticks, setting the length of the ticks with `ticklen` and the color with `tickcolor`. We've also turned on grid lines for the x-axis minor ticks using `showgrid`.
+
+Note: Minor ticks and grid lines are not currently supported on color bars, ternary plots, polar charts, geo plots, or on multi-categorical, log, or 3D axes.
```
`log` is supported. So you could remove it from the list here.

alexcjohnson replied:
But `category` axes also don't support minor.

archmoj replied:
> But `category` axes also don't support minor.

`category` is supported e.g. on [ticklabelposition-5 mock](https://github.com/plotly/plotly.js/blob/master/test/image/mocks/ticklabelposition-5.json).

nicolaskruchten replied:
That mock doesn't have minors on category axes... ? https://github.com/plotly/plotly.js/blob/master/test/image/baselines/ticklabelposition-5.png

nicolaskruchten replied:
I'd be more interested in how this works with `multicategory` ;)

LiamConnors commented on doc/python/axes.md:
```diff
@@ -380,6 +380,30 @@ fig.update_layout(title_text="Apple Stock Price")
+fig = px.scatter(df, x="total_bill", y="tip", color="sex")
+
+
+fig.update_xaxes(minor = {"ticklen": 6, "tickcolor": "black", "showgrid": True})
```
Ah, I wasn't aware. Thanks. Updated it now.

archmoj commented on doc/python/axes.md:
```diff
@@ -380,6 +380,30 @@ fig.update_layout(title_text="Apple Stock Price")
+
+You can position and style minor ticks on a Cartesian axis using `minor`. This takes a `dict` of properties to apply to minor ticks. Available properties include: `tickmode`, `tickvals`, `tickcolor`, `ticklen`, `tickwidth`, `dtick`, `tick0`, `nticks`, `ticks`, `showgrid`, `gridcolor`, `griddash`, and `gridwidth`.
+
+In the following example, we add minor ticks to the x-axis and then to the y-axis. For the y-axis we add ticks on the outside: `ticks="outside"`. On the x-axis we've specified some additional properties to style the minor ticks, setting the length of the ticks with `ticklen` and the color with `tickcolor`. We've also turned on grid lines for the x-axis minor ticks using `showgrid`.
```
By default ticks if being visible are located *outside*. 
I suggest on the y-axis we demo how they could be placed *inside*.

LiamConnors replied:
Good point. I've changed it to inside for the y-axis

archmoj commented on doc/python/axes.md:
```diff
@@ -464,6 +488,21 @@ fig.update_yaxes(showgrid=True, gridwidth=1, gridcolor='LightPink')
 
+_new in 5.8_
+
+By default grid lines are `solid`. Set the `griddash` property to change this style. In this example we display the x-axis grid lines as `dot`. It can also be set to `dash`, `longdash`, `dashdot`, or `longdashdot`.
```
In addition to those predefined line types, one could use custom values for `griddash` e.g. `"10px"` | more info: https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-dasharray

LiamConnors replied:
@archmoj, should it work like this? That throws an error
```fig.update_xaxes(showgrid=True, gridwidth=1, gridcolor='LightPink', griddash='10px')```

archmoj replied:
What is the error message?
Here is a [codepen](https://codepen.io/MojtabaSamimi/pen/KKQdKGL?editors=0010) you could may adapt which also illustrate the case of "monthly period labels with weekly minor ticks".

LiamConnors replied:
Created an issue for that https://github.com/plotly/plotly.py/issues/3710

LiamConnors replied:
Thanks @archmoj I've added an  example for "monthly period labels with weekly minor ticks".
